src/invariant/EventsAreConsistentWithEntryDiffs.cpp

```diff
@@ -0,0 +1,1264 @@
+// Copyright 2025 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "invariant/EventsAreConsistentWithEntryDiffs.h"
+#include "crypto/SHA.h"
+#include "invariant/InvariantManager.h"
+#include "ledger/LedgerTxn.h"
+#include "main/Application.h"
+#include "util/GlobalChecks.h"
+#include <fmt/format.h>
+#include <numeric>
+
+/*
+We're adding the signed 128 bit integer version of lib/util/uint128_t.h here to
+prevent exposing it to the protocol. uint128_t.h has deviated from the version
+in the repo mentioned which is why this is kept separate.
+
+An signed 128 bit integer type for C++
+
+https://github.com/zhanhb/int128
```

Comment:
I imported the signed version of the unsigned 128 bit integer library we already have in [core](https://github.com/stellar/stellar-core/blob/master/lib/util/uint128_t.h). The core version was modified enough that I had difficulty adding signed support back in, so I just copied the original library and removed unsigned support. I know Nico made some changes to uint128_t for safety, so those need to be made here as well. 

This isn't ideal, but not sure if there's a better solution. I had a spike that adapted the invariant to use uint128_t, but that was too messy.